src/zerocash/Address.cpp

```diff
@@ -11,26 +11,14 @@
  * @copyright  MIT license (see LICENSE file)
  *****************************************************************************/
 
-#include <cryptopp/osrng.h>
-using CryptoPP::AutoSeededRandomPool;
-
-#include <cryptopp/eccrypto.h>
-using CryptoPP::ECP;
-using CryptoPP::ECIES;
-
-#include <cryptopp/oids.h>
-namespace ASN1 = CryptoPP::ASN1;
-
-#include <cryptopp/filters.h>
-using CryptoPP::StringSink;
-using CryptoPP::StringStore;
+#include "zcash/NoteEncryption.hpp"
 
 #include "Zerocash.h"
 #include "Address.h"
 
 namespace libzerocash {
 
-PrivateAddress::PrivateAddress(const std::vector<unsigned char> a_sk, const std::string sk_enc) {
+PrivateAddress::PrivateAddress(const uint256 &a_sk, const uint256 &sk_enc) {
```

Comment:
[See my other comment.](https://github.com/zcash/zcash/pull/827#discussion_r59966980) `uint256` is effectively just an `unsigned char[32]` and has no endianness properties. You cannot do arithmetic on it. In fact, we use it quite often throughout our current implementation with the existing circuit.